crates/oxc_parser/src/lexer/mod.rs

```diff
@@ -414,25 +404,28 @@ impl<'a> Lexer<'a> {
                         self.private_identifier(builder)
                     }
                 }
-                '@' => Kind::At,
-                '\\' => {
+                // placeholder for '\'
+                Kind::Unknown => {
                     builder.force_allocation_without_current_ascii_char(self);
                     self.identifier_unicode_escape_sequence(&mut builder, true);
                     self.identifier_name_or_keyword(builder)
                 }
-                c if is_identifier_start(c) => {
-                    builder.push_matching(c);
-                    self.identifier_name_or_keyword(builder)
-                }
-                c if is_irregular_whitespace(c) => Kind::WhiteSpace,
-                c if is_irregular_line_terminator(c) => {
-                    self.current.token.is_on_new_line = true;
-                    Kind::NewLine
-                }
-                _ => {
-                    self.error(Diagnostic::InvalidCharacter(c, self.unterminated_range()));
-                    Kind::Undetermined
-                }
+                Kind::Undetermined => match c {
+                    c if is_irregular_whitespace(c) => Kind::WhiteSpace,
+                    c if is_irregular_line_terminator(c) => {
+                        self.current.token.is_on_new_line = true;
+                        Kind::NewLine
+                    }
+                    c if unicode_id_start::is_id_start(c) => {
```

Comment:
Note to self: this functions checks for ascii again, I need to publish a new version for unicode_id_start